clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/chunking/AbstractAvroChunkingAdapter.java

```diff
@@ -127,6 +128,36 @@ public T get(
         manifestContainer);
   }
 
+  public ByteBufferValueRecord<T> getWithSchemaId(
+      AbstractStorageEngine store,
+      int partition,
+      ByteBuffer key,
+      boolean isChunked,
+      T reusedValue,
+      BinaryDecoder reusedDecoder,
+      ReadResponse response,
+      int readerSchemaId,
```

Comment:
Is there any chance of getting rid of the superfluous params in this new API?

The `readerSchemaId` one, for example, causes some lookups at the call site, which I would prefer to avoid if the param is not even used...

The passed in `response` param is also null, so we could get rid of the conditional code related to it later on in the implementation...

Reply:
Good idea.

Reply:
Updated, readerSchemaId I guess is used. intellij led me to believe another implementation was getting used where it was getting thrown away, but it turns out it is used by the AbsctractAvroChunkingAdapter implementation (where ChunkingAdapter drops it on the floor).  Stepping through it with the debugger confirmed it.  But I did drop some other parameters from the signature.

Reply:
Hmm ok... but in our case it won't go through the code path where it's used, and we could just hardcode -1 into it rather than perform the lookup, right?

Reply:
Huh.  I did just hardcode it and you are right.  And now I think I get why.  The RawBytesDecoder shouldn't be doing any translating, it just returns raw bytes (hence the name).  It just needed it because of the interface contract. So removed it.